src/bindgen/language_backend/java_jna.rs

```diff
@@ -0,0 +1,860 @@
+                        PrimitiveType::Char => "ByteByReference",
+                        PrimitiveType::SChar => "ByteByReference",
+                        PrimitiveType::UChar => "ByteByReference",
+                        PrimitiveType::Char32 => "Pointer",
+                        PrimitiveType::Float => "FloatByReference",
+                        PrimitiveType::Double => "DoubleByReference",
+                        PrimitiveType::VaList => "PointerByReference",
+                        PrimitiveType::PtrDiffT => "PointerByReference",
+                        PrimitiveType::Integer { kind, .. } => {
+                            match kind {
+                                IntKind::Short => "ShortByReference",
+                                IntKind::Int => "IntByReference",
+                                IntKind::Long => "NativeLongByReference",
+                                IntKind::LongLong => "LongByReference",
+                                IntKind::SizeT => "NativeLongByReference", // TODO probably not right
+                                IntKind::Size => "NativeLongByReference", // TODO probably not right
+                                IntKind::B8 => "ByteByReference",
+                                IntKind::B16 => "ShortByReference",
+                                IntKind::B32 => "IntByReference",
+                                IntKind::B64 => "LongByReference",
+                            }
+                        }
+                    };
+                    write!(out, "{typ}")
+                }
+                Type::Array(_, _) => out.write("Pointer"),
+                Type::FuncPtr { .. } => out.write("CallbackReference"),
+            },
+
+            Type::Path(path) => {
+                write!(out, "{}", path.export_name())
+            }
+            Type::Primitive(primitive) => {
+                //https://github.com/java-native-access/jna/blob/master/www/Mappings.md
+                let typ = match primitive {
+                    PrimitiveType::Void => "void",
+                    PrimitiveType::Bool => "boolean",
+                    PrimitiveType::Char => "byte",
+                    PrimitiveType::SChar => "byte",
+                    PrimitiveType::UChar => "byte",
+                    PrimitiveType::Char32 => "char",
+                    PrimitiveType::Float => "float",
+                    PrimitiveType::Double => "double",
+                    PrimitiveType::VaList => "Pointer",
+                    PrimitiveType::PtrDiffT => "Pointer",
+                    PrimitiveType::Integer { kind, .. } => {
+                        match kind {
+                            IntKind::Short => "short",
+                            IntKind::Int => "int",
+                            IntKind::Long => "NativeLong",
+                            IntKind::LongLong => "long",
+                            IntKind::SizeT => "NativeLong", // TODO probably not right
```

Comment:
Def. not right. Should be "pointer" most likely.

Reply:
Fixed, now using a dedicated JNA IntegerType using size SIZE_T_SIZE